src/js/angular/repositories/ontop-repo-constants.js

```diff
@@ -0,0 +1,240 @@
+        }
+    });
+
+    modalInstance.result.then(function (data) {
+        // send data to backend
+        RepositoriesRestService.updateRepositoryFileContent(data.fileLocation, data.content).success(function(result) {
+            $scope.ontopRepoFileNames[file] = getFileName(result.fileLocation);
+            $scope.repositoryInfo.params[file].value = result.fileLocation;
+        }).error(function (error) {
+            const msg = getError(error);
+            toastr.error(msg, 'Error');
+        })
+    });
+}
+
+export const uploadRepoFile = function (files, param, Upload, $scope, toastr) {
+    if (files && files.length) {
+        $scope.uploadFile = files[0];
+        $scope.uploadFileLoader = true;
+        Upload.upload({
+            url: 'rest/repositories/uploadFile',
+            data: {uploadFile: $scope.uploadFile}
+        })
+            .success(function (data) {
+                if (!data.success) {
+                    toastr.error(data.errorMessage);
+                } else {
+                    $scope.ontopRepoFileNames[param] = $scope.uploadFile.name;
+                    $scope.repositoryInfo.params[param].value = data.fileLocation;
+                }
+                $scope.uploadFileLoader = false;
+            }).error(function (data) {
+            const msg = getError(data);
+            toastr.error(msg, 'Error');
+            $scope.uploadFile = '';
+            $scope.uploadFileLoader = false;
+        });
+    }
+};
+
+export const getFileName = function(path) {
+    let lastIdx = path.lastIndexOf('/');
+    if (lastIdx === -1) {
+        lastIdx = path.lastIndexOf('\\');
+    }
+    let name = path;
+    if (lastIdx !== -1) {
+        name = name.substring(lastIdx + 1);
+    }
+    return name;
+};
+
+export const concatURL = function(labelName, $scope) {
+    if (labelName === 'hostName' || labelName === 'port' || labelName === 'databaseName') {
+        let result = $scope.selectedDriver.urlTemplate;
+        if ($scope.selectedDriver.jdbc.hostName) {
+            if ($scope.selectedDriver.jdbc.port) {
+                result = result.replace('{hostport}',
+                    $scope.selectedDriver.jdbc.hostName + ':' + $scope.selectedDriver.jdbc.port);
+            } else {
+                result = result.replace('{hostport}', $scope.selectedDriver.jdbc.hostName);
+            }
+        }
+
+        if ($scope.selectedDriver.jdbc.databaseName) {
+            result = result.replace('{database}', $scope.selectedDriver.jdbc.databaseName);
+        }
+
+        $scope.selectedDriver.jdbc.url = result;
+    }
+}
+
+export const getInputType = function (labelName) {
+    switch (labelName) {
+        case 'password':
+            return 'password';
+        case 'port':
+            return 'number';
+        default:
+            return 'text';
+    }
+}
+
+export const checkForRequiredOntopFiles = function ($scope, RepositoriesRestService, toastr) {
+    if ($scope.repositoryInfo.type === ONTOP_TYPE) {
```

Comment:
For convenience it's better if a method returns a promise in some cases, to make it always return a promise. Otherwise the method client would need to wrap the result in promise again or to check if the result is promise or not.